File: frontend/src/services/adminPurchaseOrderService.js
import api from './api';

/**
 * Service Layer - Admin APIs for generating and viewing purchase orders
 */

/**
 * Get all purchase orders across all brands.
 * @returns {Promise<Array>} array of PurchaseOrderDto
 */
export const getAllPurchaseOrders = () => 
  api.get('/api/admin/purchase-orders').then((res) => res.data);

/**
 * Create a new purchase order
 * @param {Object} data format { brandId, items: [{ productId, quantity }] }
 * @returns {Promise<Object>} The created order
 */
export const createPurchaseOrder = (data) =>
  api.post('/api/admin/purchase-orders', data).then((res) => res.data);
/**
 * Confirm delivery and update inventory stock
 * @param {number} orderId Purchase order ID
 * @returns {Promise<Object>} The updated order
 */
export const confirmDeliveryAndStock = (orderId) =>
  api.put(`/api/admin/purchase-orders/${orderId}/confirm`).then((res) => res.data);

File: backend/src/main/java/com/urbanfresh/service/AdminPurchaseOrderService.java
package com.urbanfresh.service;

import java.util.List;
import com.urbanfresh.dto.request.CreatePurchaseOrderRequest;
import com.urbanfresh.dto.response.PurchaseOrderDto;

/**
 * Service Layer - Manages admin-facing purchase order creation and tracking.
 */
public interface AdminPurchaseOrderService {

    /**
     * Create a new purchase order for a specific brand's products.
     * @param request the purchase order details.
     * @return the created purchase order DTO.
     */
    PurchaseOrderDto createPurchaseOrder(CreatePurchaseOrderRequest request);

    /**
     * Retrieve all purchase orders globally for admins.
     * @return list of all purchase orders.
     */
    List<PurchaseOrderDto> getAllPurchaseOrders();

    /**
     * Confirm a purchase order has been received, updating inventory stock.
     * @param orderId the ID of the purchase order.
     * @param adminUsername the username of the admin performing the action.
     * @return updated purchase order DTO.
     */
    PurchaseOrderDto confirmDeliveryAndStock(Long orderId, String adminUsername);
}